src/modules/esl-panel-group/core/esl-panel-group.ts

```diff
@@ -131,64 +117,80 @@ export class ESLPanelGroup extends ESLBaseElement {
+  /** @returns action params config that's used (inherited) by controlled {@link ESLPanel}s */
   public get panelConfig(): PanelActionParams {
     return {
       noCollapse: !this.shouldCollapse || (this.currentMode === 'tabs')
     };
   }
 
+  /** @returns merged panel action params for show/hide requests from the group */
+  protected mergeActionParams(...params: PanelActionParams[]): PanelActionParams {
+    return Object.assign({initiator: 'group', activator: this}, ...params);
+  }
+
   /** Condition-guard to check if the passed target is a Panel that should be controlled by the Group */
   public includesPanel(target: any): target is ESLPanel {
     if (!(target instanceof ESLPanel)) return false;
     return target.$group === this;
   }
 
-  /** Show all panels besides excluded ones */
-  public showAll(excluded: ESLPanel[] = []) {
-    this.$panels.forEach((el) => !excluded.includes(el) && el.show(this.actionParams));
+  /** Shows all panels besides excluded ones */
+  public showAll(excluded: ESLPanel[] = [], params: PanelActionParams = {}) {
+    this.$panels.forEach((el) => !excluded.includes(el) && el.show(this.mergeActionParams(params)));
   }
-  /** Hide all active panels besides excluded ones */
-  public hideAll(excluded: ESLPanel[] = []) {
-    this.$activePanels.forEach((el) => !excluded.includes(el) && el.hide(this.actionParams));
+  /** Hides all active panels besides excluded ones */
+  public hideAll(excluded: ESLPanel[] = [], params: PanelActionParams = {}) {
+    this.$activePanels.forEach((el) => !excluded.includes(el) && el.hide(this.mergeActionParams(params)));
   }
-  /** Toggle all panels by predicate */
-  public toggleAllBy(shouldOpen: (panel: ESLPanel) => boolean) {
-    this.$panels.forEach((panel) => panel.toggle(shouldOpen(panel), this.actionParams));
+  /** Toggles all panels by predicate */
+  public toggleAllBy(shouldOpen: (panel: ESLPanel) => boolean, params: PanelActionParams = {}) {
+    this.$panels.forEach((panel) => panel.toggle(shouldOpen(panel), this.mergeActionParams(params)));
   }
 
-  /** Reset to default state applicable to the current mode */
+  /** Resets to default state applicable to the current mode */
   public reset() {
     ESLPanel.registered.then(() => {
-      if (this.currentMode === 'open') this.toggleAllBy(() => true);
+      if (this.currentMode === 'open') this.toggleAllBy(() => true, this.resetParams);
       if (this.currentMode === 'tabs' || (this.currentMode === 'accordion' && this.accordionGroup === 'single')) {
         const $activePanel = this.$panels.find((panel) => panel.initiallyOpened);
-        this.toggleAllBy((panel) => panel === $activePanel);
+        this.toggleAllBy((panel) => panel === $activePanel, this.resetParams);
       }
       if (this.currentMode === 'accordion' && this.accordionGroup === 'multiple') {
-        this.toggleAllBy((panel) => panel.initiallyOpened);
+        this.toggleAllBy((panel) => panel.initiallyOpened, this.resetParams);
```

Comment:
The function `(panel) => panel.initiallyOpened` does not seem to be used in the singular. Perhaps these functions should be defined explicitly?

Reply:
Currently, this predicate is super simple and used just twice only in the bounds of the `reset` method.
Let's keep it simple for now.